doc/source_debugging.md

```diff
@@ -0,0 +1,75 @@
+# WAMR source debugging
+
+WAMR supports source level debugging based on DWARF, source map is not supported.
```

Comment:
Should we add more details for DWARF and source map, developer may be confused, e.g.:
WAMR supports source level debugging based on DWARF (normally used in C/C++/Rust), source map (normally used in JavaScript/TypeScript) is not supported.

Reply:
Done